modules/management-console-agent/src/main/java/org/apache/ignite/agent/processor/action/DistributedActionProcessor.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.agent.processor.action;
+
+import java.util.UUID;
+import org.apache.ignite.IgniteAuthenticationException;
+import org.apache.ignite.agent.ManagementConsoleProcessor;
+import org.apache.ignite.agent.action.Session;
+import org.apache.ignite.agent.action.SessionRegistry;
+import org.apache.ignite.agent.dto.action.InvalidRequest;
+import org.apache.ignite.agent.dto.action.JobResponse;
+import org.apache.ignite.agent.dto.action.Request;
+import org.apache.ignite.agent.dto.action.ResponseError;
+import org.apache.ignite.agent.dto.action.TaskResponse;
+import org.apache.ignite.agent.ws.WebSocketManager;
+import org.apache.ignite.cluster.ClusterGroup;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.processors.GridProcessorAdapter;
+import org.apache.ignite.internal.processors.security.OperationSecurityContext;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+import static org.apache.ignite.agent.StompDestinationsUtils.buildActionJobResponseDest;
+import static org.apache.ignite.agent.StompDestinationsUtils.buildActionTaskResponseDest;
+import static org.apache.ignite.agent.dto.action.Status.FAILED;
+import static org.apache.ignite.agent.utils.AgentUtils.getErrorCode;
+
+/**
+ * Distributed action service.
+ */
+public class DistributedActionProcessor extends GridProcessorAdapter {
+    /** Authenticate action name. */
+    public static final String AUTHENTICATE_ACTION_NAME = "SecurityActions.authenticate";
+
+    /** Websocket manager. */
+    private final WebSocketManager mgr;
+
+    /** Session registry. */
+    private final SessionRegistry sesRegistry;
+
+    /** Local node consistent id. */
+    private final String locNodeConsistentId;
+
+    /**
+     * @param ctx Context.
+     */
+    public DistributedActionProcessor(GridKernalContext ctx) {
+        super(ctx);
+
+        this.mgr = ((ManagementConsoleProcessor)ctx.managementConsole()).webSocketManager();
+        this.sesRegistry = ((ManagementConsoleProcessor)ctx.managementConsole()).sessionRegistry();
+        this.locNodeConsistentId = String.valueOf(ctx.grid().localNode().consistentId());
+    }
+
+    /**
+     * @param req Request.
+     */
+    public void onActionRequest(Request req) {
+        if (log.isDebugEnabled())
+            log.debug("Received request: [sessionId=" + req.getSessionId() + ", reqId=" + req.getId() + "]");
+
+        ClusterGroup grp = F.isEmpty(req.getNodeIds())
+            ? ctx.grid().cluster().forServers()
```

Comment:
Is this correct? @akuznetsov-gridgain As I remember we need to send job to client node in case of DR